core/src/main/java/org/apache/iceberg/rest/RESTSerializers.java

```diff
@@ -87,7 +88,41 @@ public static void registerAll(ObjectMapper mapper) {
     mapper.registerModule(module);
   }
 
-  public static class UpdateRequirementDeserializer extends JsonDeserializer<UpdateRequirement> {
+  // https://github.com/FasterXML/jackson-databind/commit/da339636d6a1db6fd618f1e59816fab3bb6d38d4
+  // added JacksonException to the throws declaration of JsonDeserializer#deserialize*(...)
+  // The purpose of this class is so that we don't propagate this API-breaking throws-change to any
+  // Iceberg clients
```

Comment:
How does this avoid a breaking change? I don't see anything that catches `JacksonException`, which is an `IOException`. Since it is a subclass of an already thrown exception, what was the problem?

Reply:
Basically RevAPI complains that `throws java.io.IOException` changed to `throws java.io.IOException, com.fasterxml.jackson.core.JacksonException` due to the Jackson version bump. 
I think RevAPI doesn't care that `JacksonException` itself is a subclass of `IOException` and just complains about the signature change.

That's why we're introducing a middle class that only defines `throws java.io.IOException` on those methods. Alternatively, we could have added those RevAPI breaks as accepted breaks but that didn't seem like the better approach to me.

Reply:
Can we suppress the revapi issue to avoid adding unnecessary code?

Reply:
I don't think we can suppress it, we could only accept the API break by adding it to the rev api file